cmd/up.go

```diff
@@ -249,9 +249,12 @@ func (up *UpContext) Execute(isRetry bool) error {
 		return err
 	}
 
-	up.Sy, err = syncthing.NewSyncthing(up.Namespace, up.Deployment.Name, devList)
-	if err != nil {
-		return err
+	runSync := up.Dev.Swap.Deployment.Container == devList[0].Swap.Deployment.Container
+	if runSync {
+		up.Sy, err = syncthing.NewSyncthing(up.Namespace, up.Deployment.Name, devList)
+		if err != nil {
+			return err
```

Comment:
At this level, I wouldn't return the err directly, since this will be displayed to the user. Maybe return a more generic error, like fmt.Errorf("couldn't start file synchronization, please try again), and log.Info the real error

I'd apply this pattern to any error at this level, since it's the one before the user (in the inner layers it makes sense to just bubble it up).